firewood/src/merkle.rs

```diff
@@ -1293,6 +1293,15 @@ impl<S: ShaleStore<Node> + Send + Sync, T> Merkle<S, T> {
         last_key: Option<K>,
         limit: Option<usize>,
     ) -> Result<Option<api::RangeProof<Vec<u8>, Vec<u8>>>, api::Error> {
+        if let (Some(k1), Some(k2)) = (&first_key, &last_key) {
+            if k1.as_ref() > k2.as_ref() {
```

Comment:
I think you can do `if k1.as_ref() > k2.as_ref()` by adding the `PartialOrd` trait bound to `K`. Up to you though. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3f9f79a8b89f9a65c6eb838b2402ee78

Reply:
Do you mean we can do `if k1 > k2` ?

Reply:
Yes, you could make it so keys can directly be compared if you change `KeyType` to be `PartialOrd` as well. This would restrict custom types from being used as keys unless they implement `PartialOrd` though; I think calling `as_ref` first is probably better than forcing everyone to implement `PartialOrd` for just this one use case.